src/topic_training/src/Begin_Here.md

```diff
@@ -0,0 +1,262 @@
+
+### Python
+
+A basic setup for ROS topics in Python would look something like this:
+
+``` Python
+rospy.init_node('Demo', anonymous=True)
+rospy.Subscriber('plot', Float64, callback)
+pub = rospy.Publisher('chatter', String, queue_size=QUEUE_SIZE)
+```
+
+To break this down,
+
+``` Python
+rospy.init_node('Demo', anonymous=True)
+```
+Initialize ROS and specify the name of the node (which is the name of the file in this case).
+
+``` Python
+rospy.Subscriber('plot', Float64, callback)
+```
+Initialize the Subscriber. State the name of the topic (`plot`), the message type we will be publishing (float, which is actually in the class std_msgs.msg.Float64), and the function that will be called when a message is received (callBack(data)).
+
+``` Python
+pub = rospy.Publisher('chatter', String, queue_size=QUEUE_SIZE)
+```
+Initialize the Publisher. State the name of the topic (`chatter`), the message type we will be publishing (String, which is actually in the class std_msgs.msg.String), and the queue size (1). The queue size is the maximum number of messages kept as a buffer before throwing away old ones if we are publishing too quickly.
+
+The callback function for a Subscriber looks something like this:
+
+``` Python
+def callback(data: Float64) -> None:
+    global pi 
+    pi = data.data
+```
+This function will get called when a new message has arrived on the topic `plot`. The message received by the Subscriber, sent by the Publisher, is denoted as `msg`.
+
+Publishing a message on a topic would look something like this:
+
+``` Python
+pi_str = "currently at %s" % pi
+pub.publish(hello_str)
+```
+
+To break this down,
+
+``` Python
+pi_str = "currently at %s" % pi
+```
+Create the message.
+
+``` Python
+pub.publish(hello_str)
+```
+Publish the message using the previously initialized Publisher `pub`.
+
+If you have glanced at the C++ example code, you may have noticed the function `ros::spinOnce()` and its differences with `ros::spin()`. Python only has `rospy.spin()`, which can be completely replaced by a for loop, which is why you do not see it in the Python example code.
+
+## Running a Publisher and Subscriber
+Before you are able to run the example code, find the CMakeLists.txt file that is outside of the src directory but still inside of the topic_training directory. This file will include the the configurations, builds, and installs required to run your program as well as the files you will be running. Do not change anything in the file as it should already be set up for the demo files to work. 
+
+The `find_package`, `catkin_package`, and `include_directories` specifications are required for both the C++ and Python files to work. These provide the packages and directories needed for running ROS and ROS topics. 
```

Comment:
I'm not sure `include_directories` is needed here.

Reply:
When I uncommented it out, CMake failed and gave me the error `pp:1:10: fatal error: ros/init.h: No such file or directory
    1 | #include "ros/init.h"`.

Reply:
Ok, it can stay.  Sorry.